typing_inspect.py

```diff
@@ -116,6 +116,24 @@ class MyClass(Tuple[str, int]):
     return type(tp) is TupleMeta
 
 
+def is_optional_type(tp):
+    """Returns `True` if the type is `type(None)`, or is a direct `Union` to `type(None)`, such as `Optional[T]`.
+
+    WARNING: this method might return a misleading `False` if
```

Comment:
I don't think this requires to be a **WARNING**, just make it a `NOTE`. Also it is not clear that it is a deliberate decision. If the default behaviour is to do all the checks in type variables etc. it is hard to opt-out.

Instead of calling this "misleading", better mention that if one wants to look into type variables, there are two functions: `get_constraints()` and `get_bound()`.

Reply:
ok, I changed this, let me know if that's ok